targets/TARGET_RDA/TARGET_UNO_91H/rtc_api.c

```diff
@@ -0,0 +1,95 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "rtc_api.h"
+
+#if DEVICE_RTC
+
+#include "us_ticker_api.h"
+
+#define RTC_TIMER_INITVAL_REG       (RDA_SCU->FTMRINITVAL)
+#define RTC_TIMER_TIMESTAMP_REG     (RDA_SCU->FTMRTS)
+
+#define RTC_TIMER_CLOCK_SOURCE      (32768)
+
+static time_t sw_timebase = 0U;
+static time_t sw_timeofst = 0U;
+static uint32_t round_ticks = 0U;
+static uint32_t remain_ticks = 0U;
+static int is_rtc_enabled = 0;
+
+void rtc_init(void)
+{
+    uint32_t start_time;
+    /* Make sure us_ticker is running */
+    start_time = us_ticker_read();
```

Comment:
why is this needed here? I dont see us ticker used anywhere in rtc

Reply:
RTC get timestamp from RTC_TIMER_TIMESTAMP_REG which be shared with us_ticker, us_ticker_read will ensure it has worked.